gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java

```diff
@@ -64,6 +65,10 @@ public static int eip196_perform_operation(
       int i_len,
       byte[] output) {
 
+    if (output.length < EIP196_PREALLOCATE_FOR_RESULT_BYTES) {
+      return EIP196_ERR_CODE_INVALID_OUTPUT_LENGTH;
+    }
+
```

Comment:
This is safe for eip196_perform_operation, which is our 'contract'.  The static native entrypoints are still public and potentially unsafe.  

I will approve, and leave it to your discretion to add some javadoc to the public static native entrypoints (that incorrect output length may lead to a jvm crash)

Reply:
made them private [`3dfd978` (#301)](https://github.com/hyperledger/besu-native/pull/301/commits/3dfd9781148a9487e19ac52b74b52da17daeca45)

and mentioned breaking change, but I believe our lib consumers are well known so I don't expect this to be a problem.